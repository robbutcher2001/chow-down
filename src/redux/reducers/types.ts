export interface Message {
    user: string
    message: string
}

export interface HelloState {
    hello_messages: Message[]
}

export const SAY_HELLO = 'SAY_HELLO'
export const DELETE_HELLO = 'DELETE_HELLO';

interface SayHelloAction {
    type: typeof SAY_HELLO
    rob_payload: Message
}

interface DeleteHelloAction {
    type: typeof DELETE_HELLO
}

export type SayHelloTypes = SayHelloAction | DeleteHelloAction