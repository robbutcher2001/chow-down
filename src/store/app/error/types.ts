import { Action } from 'redux';

export enum ErrorActionTypes {
  UNEXPECTED_SERVER_ERROR = '@@common/UNEXPECTED_SERVER_ERROR',
  UNEXPECTED_RESPONSE = '@@common/UNEXPECTED_RESPONSE',
  CLEAR_ERROR = '@@common/CLEAR_ERROR'
}

export interface ErrorState {
  readonly message: string
}

export interface ErrorMessageApiResponse {
  message: string
}

export interface ErrorApiResponse extends Action {
  type: ErrorActionTypes,
  json: ErrorMessageApiResponse
}