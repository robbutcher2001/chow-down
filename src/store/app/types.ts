import { Action } from 'redux';

export enum AppActionTypes {
    UNEXPECTED_SERVER_ERROR = '@@common/UNEXPECTED_SERVER_ERROR',
    UNEXPECTED_RESPONSE = '@@common/UNEXPECTED_RESPONSE',
    CLEAR_ERROR = '@@common/CLEAR_ERROR'
}

export interface AppState {
    readonly error: string
}

export interface ErrorMessageApiResponse {
    message: string
}

export interface ErrorApiResponse extends Action {
    type: AppActionTypes,
    json: ErrorMessageApiResponse
}