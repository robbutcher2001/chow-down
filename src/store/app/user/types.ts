import { Action } from 'redux';

export enum UserActionTypes {
  SET_USER_IS_SELECTING_DAY = '@@common/SET_USER_IS_SELECTING_DAY',
  CLEAR_USER_IS_SELECTING_DAY = '@@common/CLEAR_USER_IS_SELECTING_DAY'
}

export interface UserState {
  readonly selectedDay: string
}

export interface UserAction extends Action {
  type: UserActionTypes,
  day: string
}