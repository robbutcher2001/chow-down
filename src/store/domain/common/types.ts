export enum CommonActionTypes {
    UNEXPECTED_SERVER_ERROR = '@@common/UNEXPECTED_SERVER_ERROR',
    UNEXPECTED_RESPONSE = '@@common/UNEXPECTED_RESPONSE'
}