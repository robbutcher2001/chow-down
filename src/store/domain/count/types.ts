import { Action } from 'redux';

export enum CountActionTypes {
    INCREMENT_REQUEST = '@@count/INCREMENT_REQUEST',
    INCREMENT_SUCCESS = '@@count/INCREMENT_SUCCESS',
    INCREMENT_FAILURE = '@@count/INCREMENT_FAILURE',
    DECREMENT_REQUEST = '@@count/DECREMENT_REQUEST',
    DECREMENT_SUCCESS = '@@count/DECREMENT_SUCCESS',
    DECREMENT_FAILURE = '@@count/DECREMENT_FAILURE'
}

export interface CountState {
    readonly value: number
}

export interface CountApiRequest extends Action {
    type: CountActionTypes,
    payload: object
}

export interface CountSuccessApiResponse extends Action {
    type: CountActionTypes,
    json: object
}

export interface CountFailureApiResponse extends Action {
    type: CountActionTypes,
    reason: string
}

export type CountApiResponse = CountSuccessApiResponse | CountFailureApiResponse;