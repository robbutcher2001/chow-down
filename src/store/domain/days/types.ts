import { Action } from 'redux';

export enum DayActionTypes {
  GET_DAYS_REQUEST = '@@days/GET_REQUEST',
  GET_DAYS_SUCCESS = '@@days/GET_SUCCESS',
  GET_DAYS_FAILURE = '@@days/GET_FAILURE',
  PUT_DAYS_REQUEST = '@@days/PUT_REQUEST',
  PUT_DAYS_SUCCESS = '@@days/PUT_SUCCESS',
  PUT_DAYS_FAILURE = '@@days/PUT_FAILURE'
}

//TODO: maybe combine these with ~/store/domain/recipes
interface RecipeIngredient {
  quantity: number,
  unitSingularName: string,
  unitPluralName: string,
  ingredientName: string
}

interface Recipe {
  title: string,
  rating: number,
  image: string,
  ingredients: RecipeIngredient[]
}

export interface Day {
  date: string,
  recipeId?: string,
  recipe?: Recipe
}

export interface DaysState {
  readonly failure: string,
  readonly days: Day[]
}

export interface GetDaysApiRequest extends Action {
  type: DayActionTypes,
  from: string,
  to: string
}

export interface PutDayApiRequest extends Action {
  type: DayActionTypes,
  day: Day
}

export interface GetDaysSuccessApiResponse extends Action {
  type: DayActionTypes,
  days: Day[]
}

export interface PutDaySuccessApiResponse extends Action {
  type: DayActionTypes,
  day: Day
}

export interface DaysFailureApiResponse extends Action {
  type: DayActionTypes,
  code: number,
  json: object
}

export type GetDaysApiResponse = GetDaysSuccessApiResponse | PutDaySuccessApiResponse | DaysFailureApiResponse;