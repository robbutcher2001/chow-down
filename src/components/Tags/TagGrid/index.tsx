import React from 'react';

import styled from 'styled-components';

import { Tag } from '../../../store/domain/tags/types';

import TagComponent from '../Tag';
import { NegativeBox } from '../../MessageBox';

interface TagGridProps {
  isLoading: boolean,
  title: string,
  tags: Tag[]
};

const TagGrid = styled.div`
  h3 {
    color: ${props => props.theme.isDark ?
      props.theme.colour.lightestGrey :
      props.theme.colour.black
    };
  }

  ul {
    display: flex;
    flex-flow: row wrap;
    margin: 0;
    padding: 1rem 0rem 0rem 0rem;
    list-style: none;
    min-height: 120px;

    > li {
      flex: 1 1 50%;
      margin-bottom: 0.5rem;
    }
  }
`

export default (props: TagGridProps) =>
  !props.isLoading && props.tags.length === 0 ?
    <NegativeBox message='No tags yet!' /> :
    <TagGrid>
      <h3>{props.title}</h3>
      <ul className={props.isLoading ? 'spinner spinning' : 'spinner'} >
        {props.tags.map(tag =>
          <li key={tag.id}>
            <TagComponent
              $backgroundColour={tag.colours.background}
              $textColour={tag.colours.text}
            >
              {tag.name}
            </TagComponent>
          </li>
        )}
      </ul>
    </TagGrid>;