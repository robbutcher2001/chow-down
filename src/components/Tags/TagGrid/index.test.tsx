import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import { ThemeProvider } from 'styled-components';

import theme from '../../../themes';
import TagGrid from '.';
import { Tag } from '../../../store/domain/tags/types';

const mockTags: Tag[] = [{
  id: '123',
  name: 'fake_tag1',
  colours: {
    background: 'red',
    text: 'white'
  }
},
{
  id: '456',
  name: 'fake_tag2',
  colours: {
    background: 'blue',
    text: 'red'
  }
}];

test('TagGrid basic snapshot render', () => {
  const tagGrid = renderer.create(
    <Router history={createMemoryHistory()}>
      <ThemeProvider theme={theme}>
        <TagGrid isLoading={false} title='Tags' tags={mockTags} />
      </ThemeProvider>
    </Router>
  );

  expect(tagGrid.toJSON()).toMatchSnapshot();
});

test('TagGrid basic isLoading snapshot render', () => {
  const tagGrid = renderer.create(
    <Router history={createMemoryHistory()}>
      <ThemeProvider theme={theme}>
        <TagGrid isLoading={true} title='Tags' tags={mockTags} />
      </ThemeProvider>
    </Router>
  );

  expect(tagGrid.toJSON()).toMatchSnapshot();
});

test('TagGrid correct tag content assertion', () => {
  const { getByText } = render(
    <Router history={createMemoryHistory()}>
      <ThemeProvider theme={theme}>
        <TagGrid isLoading={false} title='Tags' tags={mockTags} />
      </ThemeProvider>
    </Router>
  );

  expect(getByText(/fake_tag1/).textContent).toEqual('fake_tag1');
});

test('TagGrid render multiple tags assertion', () => {
  const { getByText } = render(
    <Router history={createMemoryHistory()}>
      <ThemeProvider theme={theme}>
        <TagGrid isLoading={false} title='Tags' tags={mockTags} />
      </ThemeProvider>
    </Router>
  );

  expect(getByText(/fake_tag1/).textContent).toEqual('fake_tag1');
});

test('TagGrid correct class when not loading', () => {
  const { container } = render(
    <Router history={createMemoryHistory()}>
      <ThemeProvider theme={theme}>
        <TagGrid isLoading={false} title='Tags' tags={mockTags} />
      </ThemeProvider>
    </Router>
  );

  expect(container.querySelector('ul').classList).toContain('spinner');
  expect(container.querySelector('ul').classList).not.toContain('spinning');
});

test('TagGrid correct class when loading', () => {
  const { container } = render(
    <Router history={createMemoryHistory()}>
      <ThemeProvider theme={theme}>
        <TagGrid isLoading={true} title='Tags' tags={mockTags} />
      </ThemeProvider>
    </Router>
  );

  expect(container.querySelector('ul').classList).toContain('spinner');
  expect(container.querySelector('ul').classList).toContain('spinning');
});