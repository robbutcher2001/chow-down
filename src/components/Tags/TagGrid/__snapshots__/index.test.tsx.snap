// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagGrid basic isLoading snapshot render 1`] = `
<div
  className="sc-AxiKw hJnbkr"
>
  <h3>
    Tags
  </h3>
  <ul
    className="spinner spinning"
  >
    <li>
      <span
        className="sc-AxjAm bvsuMB"
      >
        fake_tag1
      </span>
    </li>
    <li>
      <span
        className="sc-AxjAm hgQdmQ"
      >
        fake_tag2
      </span>
    </li>
  </ul>
</div>
`;

exports[`TagGrid basic snapshot render 1`] = `
<div
  className="sc-AxiKw hJnbkr"
>
  <h3>
    Tags
  </h3>
  <ul
    className="spinner"
  >
    <li>
      <span
        className="sc-AxjAm bvsuMB"
      >
        fake_tag1
      </span>
    </li>
    <li>
      <span
        className="sc-AxjAm hgQdmQ"
      >
        fake_tag2
      </span>
    </li>
  </ul>
</div>
`;
